packages/cashier/src/components/deposit-sub-page-event-tracker/deposit-sub-page-analytics-event-tracker.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from 'react';
+import { Analytics } from '@deriv-com/analytics';
+import { useStore } from '@deriv/stores';
+
+type TProps = { deposit_category: 'crypto' | 'fiat' | 'fiat_onramp' | 'payment_agent' | 'p2p' };
+
+const DepositSubPageAnalyticsEventTracker: React.FC<TProps> = ({ deposit_category }) => {
```

Comment:
@wojciech-deriv 
yeap i was thinking about it @wojciech-deriv 
one concern that bothers me is that we should only track events on successful deposit subpages, i mean that if there is a cashier locked or any other scenario that is block  successful deposit subpage to render we should not track it. The problem is some of that scenarios are inside the paren component, smth like:
```
const PublicAgent = () => {
    // if we are calling hook here we can't know if we eventually will show `Payment agent code`
    analyticHookWithUseEffect(); 

    if(isLoading) return Loader

    if(is_smth_else) return SmthElse

    if(is_smth_else) return SmthElse

    if(is_smth_else) return SmthElse

    //here we can't put `analyticHookWithUseEffect` because we can't render hooks conditionally

    return (<div>
        // now we 100% sure that we are on the right page
        </DepositSubPageAnalyticsEventTracker> that's why i put here
        Payment agent code
    </div>)
}
```
I hope you get my concern here.

Reply:
I believe in order to use hook, we need restructure the code of parent components.